include/nlohmann/json.hpp

```diff
@@ -2540,7 +2768,7 @@ class basic_json
     @brief get a value (explicit)
 
     Explicit type conversion between the JSON value and a compatible value
-    which is [CopyConstructible](https://en.cppreference.com/w/cpp/named_req/CopyConstructible)
+    which is [Copynonstructible](https://en.cppreference.com/w/cpp/named_req/CopyConstructible)
```

Comment:
`Copynonstructible` -> `CopyConstructible`

Reply:
Right, again I agree that the `serialize_binary` parameter is a janky hack.  Again, I only put it in the make the test harness work.  Otherwise when you call:

    CHECK(result == expected);

You get a unhandled exception because `CHECK` wants to dump the object using `operator<<`, and I don't know how to override this behavior or if it's even possible to override this behavior.  But it doesn't seem like it is, the offending line seems to be found [here](https://github.com/onqtam/doctest/blob/master/doctest/parts/doctest_fwd.h#L1994)? __VA_ARGS__ is passed via operator<< to the expression decomposer?

I'm sure I'm misunderstanding this code.  But the long story short is, when I tried it previously without the janky hack I was simply unable to test the code at all!  It's ugly, and it's awful, and I really, really hate it, but I cannot figure out a way around it.

Reply:
Actually, I retract all of this.  This must have only been a problem with the old test suite.  Doctest seems to handle this just fine now.  I shall remove this flag, but leave in the `dump` parameter.